include/gridtools/storage/adapter/python_sid_adapter.hpp

```diff
@@ -37,22 +39,39 @@ namespace gridtools {
         template <size_t, class>
         struct kind {};
 
-        template <class T, size_t Dim, class Kind>
+        template <int UnitStrideDim>
+        struct transform_strides_f {
+            template <size_t I, class T, std::enable_if_t<I == UnitStrideDim, int> = 0>
+            integral_constant<pybind11::ssize_t, 1> operator()(T val) const {
+                if (val != 1)
+                    throw std::domain_error("incompatible strides, expected unit stride");
+                return {};
+            }
+
+            template <size_t I, class T, std::enable_if_t<I != UnitStrideDim, int> = 0>
+            T operator()(T val) const {
+                return val;
+            }
+        };
+
+        template <class T, size_t Dim, class Kind, int UnitStrideDim>
         struct wrapper {
             pybind11::buffer_info m_info;
 
             friend sid::simple_ptr_holder<T *> sid_get_origin(wrapper const &obj) {
                 return {reinterpret_cast<T *>(obj.m_info.ptr)};
             }
-            friend std::array<pybind11::ssize_t, Dim> sid_get_strides(wrapper const &obj) {
+            friend auto sid_get_strides(wrapper const &obj) {
                 std::array<pybind11::ssize_t, Dim> res;
                 assert(obj.m_info.strides.size() == Dim);
                 for (std::size_t i = 0; i != Dim; ++i) {
                     assert(obj.m_info.strides[i] % obj.m_info.itemsize == 0);
                     res[i] = obj.m_info.strides[i] / obj.m_info.itemsize;
                 }
-                return res;
+                return tuple_util::transform_index(
+                    transform_strides_f<UnitStrideDim>{}, tuple_util::convert_to<tuple>(res));
```

Comment:
I actually verified that it is required as otherwise the return type of `transform_index` is still `array<pybind11::ssize_t, Dim>` (which obviously does not support the desired replacement of a single element by a `integral_constant`).

Reply:
Yes. My bad.